pkg/aws/awscluster/mutate_awscluster_test.go

```diff
@@ -71,3 +72,62 @@ func TestAWSClusterPodCIDR(t *testing.T) {
 		})
 	}
 }
+
+func TestAWSClusterDescription(t *testing.T) {
+	testCases := []struct {
+		ctx  context.Context
+		name string
+
+		currentDescription string
+		expectedPatch      string
+	}{
+		{
+			// Don't default the Cluster Description if it is set
+			name: "case 0",
+			ctx:  context.Background(),
+
+			currentDescription: "My cluster",
+			expectedPatch:      "",
+		},
+		{
+			// Default the Cluster Description if it is not set
+			name: "case 1",
+			ctx:  context.Background(),
+
+			currentDescription: "",
+			expectedPatch:      aws.DefaultClusterDescription,
+		},
+	}
+	for i, tc := range testCases {
+		t.Run(strconv.Itoa(i), func(t *testing.T) {
+			var err error
+			var updatedDescription string
+
+			fakeK8sClient := unittest.FakeK8sClient()
+			mutate := &Mutator{
+				podCIDRBlock: unittest.DefaultPodCIDR,
+				k8sClient:    fakeK8sClient,
+				logger:       microloggertest.New(),
+			}
+
+			// run mutate function to default AWSCluster Description
+			var patch []mutator.PatchOperation
+			awscluster := unittest.DefaultAWSCluster()
+			awscluster.Spec.Cluster.Description = tc.currentDescription
+			patch, err = mutate.MutateDescription(awscluster)
```

Comment:
Towards https://github.com/giantswarm/giantswarm/issues/13827 we only run the mutating function as unit test here instead of going through the entire admission process